@shared/api/internal.ts

```diff
@@ -367,11 +367,16 @@ export const signFreighterTransaction = async ({
   transactionXDR: string;
   network: string;
 }): Promise<{ signedTransaction: string }> => {
-  const { signedTransaction } = await sendMessageToBackground({
+  const { signedTransaction, error } = await sendMessageToBackground({
```

Comment:
better error handling if signing fails, moves the `transactionSubmission` slice to failed

Reply:
TIL learned if the method in background `popupMessageListener` throws it returns the error in the response instead of propagating to the popup methods